### PR TITLE
Add synchronization delay section in lp ticker

### PR DESCRIPTION
**Pull request by c1728p9**
Add a section with information about low power ticker synchronization and using LOWPOWERTIMER_DELAY_TICKS to address this problem.

### CODE REVIEW

**Comment by c1728p9** [comment]
https://github.com/ARMmbed/mbed-os/pull/6536 should be merged before this is

File: docs/reference/contributing/target/lp_ticker.md [hunk 1]
```diff
@@ -36,6 +36,12 @@ Be careful around these common trouble areas when implementing this API:
 - The ticker keeps counting when it rolls over
 - The ticker interrupt fires when the compare value is set to 0 and and overflow occurs
 
+#### Handling synchronization delay
+
+Some low power tickers require multiple low power clock cycles for the compare value set by  `ticker_set_interrupt` to take effect. Further complicating this issue, a new compare value typically cannot be set until the first has taken effect. Because of this when back to back calls to `ticker_set_interrupt` are made without a delay the second call will block and violate the above requirement that `ticker_set_interrupt` completes in 20us.
+
+To meet this timing requirement targets which have this synchronization delay must define `LOWPOWERTIMER_DELAY_TICKS` to the number of low power clock cycles it takes for call to `ticker_set_interrupt` to take effect. When this value is set the timer code will prevent `lp_ticker_set_interrupt` from being called twice within that number of clock cycles. It does this by using the microsecond time to schedule the write to happen at a future date.
```
Comment (jeromecoutant) [inline docs/reference/contributing/target/lp_ticker.md]: Hi
I think LOWPOWERTIMER_DELAY_TICKS will become LPTICKER_DELAY_TICKS when feature branch will be merged ?

File: docs/reference/contributing/target/lp_ticker.md [hunk 2]
```diff
@@ -36,6 +36,12 @@ Be careful around these common trouble areas when implementing this API:
 - The ticker keeps counting when it rolls over
 - The ticker interrupt fires when the compare value is set to 0 and and overflow occurs
 
+#### Handling synchronization delay
+
+Some low power tickers require multiple low power clock cycles for the compare value that `ticker_set_interrupt` sets to take effect. Further complicating this issue, a new compare value typically cannot be set until the first has taken effect. Because of this, when you make back-to-back calls to `ticker_set_interrupt` without a delay, the second call blocks and violates the above requirement that `ticker_set_interrupt` completes in 20us.
```
Comment (AnotherButler) [inline docs/reference/contributing/target/lp_ticker.md]: Query: Who or what typically sets a new compare value?

File: docs/reference/contributing/target/lp_ticker.md [hunk 3]
```diff
@@ -36,6 +36,12 @@ Be careful around these common trouble areas when implementing this API:
...
+
+To meet this timing requirement, targets that have this synchronization delay must define `LOWPOWERTIMER_DELAY_TICKS` to the number of low power clock cycles it takes for a call to `ticker_set_interrupt` to take effect. When this value is set, the timer code prevents `lp_ticker_set_interrupt` from being called twice within that number of clock cycles. It does this by using the microsecond time to schedule the write to happen at a future date.
```
Comment (AnotherButler) [inline docs/reference/contributing/target/lp_ticker.md]: Query: Is this saying that targets define `LOWPOWERTIMER_DELAY_TICKS` as the number of low power clock cycles required for a call to `ticker_set_interrupt` to take effect? Also, in the next sentence, does "this value" refer to that same number of low power clock cycles? If so, do the targets set that value, or do you, the user?

File: docs/reference/contributing/target/lp_ticker.md [hunk 4]
```diff
@@ -36,6 +36,12 @@ Be careful around these common trouble areas when implementing this API:
 - The ticker keeps counting when it rolls over
 - The ticker interrupt fires when the compare value is set to 0 and and overflow occurs
 
+#### Handling synchronization delay
+
+Some low power tickers require multiple low power clock cycles for the compare value that `ticker_set_interrupt` sets to take effect. Further complicating this issue, a new compare value typically cannot be set until the first has taken effect. Because of this, when you make back-to-back calls to `ticker_set_interrupt` without a delay, the second call blocks and violates the above requirement that `ticker_set_interrupt` completes in 20us.
+
+To meet this timing requirement, targets that have this synchronization delay must set `LOWPOWERTIMER_DELAY_TICKS` to the number of low power clock cycles it takes for a call to `ticker_set_interrupt` to take effect. When the targets set this value, the timer code prevents `lp_ticker_set_interrupt` from being called twice within that number of clock cycles. It does this by using the microsecond time to schedule the write to happen at a future date.
```
Comment (jeromecoutant) [inline docs/reference/contributing/target/lp_ticker.md]: Hi
So LOWPOWERTIMER_DELAY_TICKS will not be renamed as LPTICKER_DELAY_TICKS ?

Comment (c1728p9) [reply docs/reference/contributing/target/lp_ticker.md]: Hi @jeromecoutant sorry for not responding earlier. I'm not sure when the feature branch is going to be merged to master, so I left this with the name on master. Once the feature branch is merged to master I'll update this for the new name.

**Pull request merged**